cmd/govim/vimstate.go

```diff
@@ -113,9 +113,9 @@ func (v *vimstate) setConfig(args ...json.RawMessage) (interface{}, error) {
 	if !vimconfig.EqualBool(v.config.HighlightDiagnostics, preConfig.HighlightDiagnostics) {
 		if v.config.HighlightDiagnostics == nil || !*v.config.HighlightDiagnostics {
 			// HighlightDiagnostics is now not on - remove existing text properties
-			v.redefineHighlights([]types.Diagnostic{})
+			v.redefineHighlights([]types.Diagnostic{}, true)
```

Comment:
Again this is another situation where I think we could end up trying to apply bad highlights. 

* User makes change
* User quickly changes config
* Diagnostics are the old diagnostics (new ones haven't arrived yet)
* Bad diagnostics get applied as highlights

Reply:
This particular case is safe since it's only removing diagnostics (passing in an empty slice).

Reply:
Ah yes, commented on wrong one.